src/nomad_simulations/schema_packages/basis_set.py

```diff
@@ -0,0 +1,642 @@
+        description="""
+        Number of terms in the local orbital.
+        """,
+    )
+
+    energy_parameter = Quantity(
+        type=np.float64,
+        shape=['n_terms'],
+        unit='joule',
+        description="""
+        Reference energy parameter for the augmented plane wave (APW) basis set.
+        Is used to set the energy parameter for each state.
+        """,
+    )  # TODO: add approximation formula from energy parameter n
+
+    energy_parameter_n = Quantity(
+        type=np.int32,
+        shape=['n_terms'],
+        description="""
+        Reference number of radial nodes for the augmented plane wave (APW) basis set.
+        This is used to derive the `energy_parameter`.
+        """,
+    )
+
+    energy_status = Quantity(
+        type=MEnum('fixed', 'pre-optimization', 'post-optimization'),
+        default='post-optimization',
+        description="""
+        Allow the code to optimize the initial energy parameter.
+        """,
+    )
+
+    differential_order = Quantity(
+        type=np.int32,
+        shape=['n_terms'],
+        description="""
+        Derivative order of the radial wavefunction term.
+        """,
+    )  # TODO: add check non-negative # ? to remove
+
+    def _get_open_quantities(self) -> set[str]:
+        """Extract the open quantities of the `APWBaseOrbital`."""
+        return {
+            k for k, v in self.m_def.all_quantities.items() if self.m_get(v) is not None
+        }
+
+    def _get_lengths(self, quantities: set[str]) -> list[int]:
+        """Extract the lengths of the `quantities` contained in the set."""
+        present_quantities = set(quantities) & self._get_open_quantities()
+        return [len(getattr(self, quant)) for quant in present_quantities]
+
+    def _of_equal_length(self, lengths: list[int]) -> bool:
+        """Check if all elements in the list are of equal length."""
+        if len(lengths) == 0:
+            return True
+        else:
+            ref_length = lengths[0]
+            return all(length == ref_length for length in lengths)
+
+    def get_n_terms(
+        self,
+        representative_quantities: set[str] = {
+            'energy_parameter',
+            'energy_parameter_n',
+            'differential_order',
+        },
+    ) -> Optional[int]:
+        """Determine the value of `n_terms` based on the lengths of the representative quantities."""
+        lengths = self._get_lengths(representative_quantities)
+        if not self._of_equal_length(lengths) or len(lengths) == 0:
+            return None
+        else:
+            return lengths[0]
+
+    def normalize(self, archive: EntryArchive, logger: BoundLogger) -> None:
+        super().normalize(archive, logger)
+
+        # enforce quantity length (will be used for type assignment)
+        new_n_terms = self.get_n_terms()
+        if self.n_terms is None:
+            self.n_terms = new_n_terms
+        elif self.n_terms != new_n_terms:
+            if logger is not None:
```

Comment:
please, take out this condition, logger should be set anyways

Reply:
You're right, this goes outside the interface specification. Done